back/app/Http/Controllers/MessageController.php

```diff
@@ -35,6 +35,17 @@ public function update(MessageRequest $request,$message_id)
         return response()->json(MessageResource::make($message), JsonResponse::HTTP_CREATED);
     }
 
+    public function updateActivated(Request $request, $message_id)
+    {
+        $message = Message::withTrashed()->findOrFail($message_id); //Vu qu'il y a un soft delete j'ai mis le withTrashed()
```

Comment:
Alors ici le withTrashed n'est pas nécessaire car on ne veut pas pouvoir activer ou désactiver un message qui est (soft) supprimé. Mais super que tu y ai pensé 👍 
```suggestion
        $message = Message::findOrFail($message_id);
```